core/http/client/src/main/java/org/eclipse/rdf4j/http/client/SPARQLProtocolSession.java

```diff
@@ -1041,6 +1044,10 @@ protected HttpResponse execute(HttpUriRequest method) throws IOException, RDF4JE
 						throw new MalformedQueryException(errInfo.getErrorMessage());
 					} else if (errInfo.getErrorType() == ErrorType.UNSUPPORTED_QUERY_LANGUAGE) {
 						throw new UnsupportedQueryLanguageException(errInfo.getErrorMessage());
+					} else if (exceptionIs(response, "ShaclSailValidationException")) {
+						throw new RepositoryException(new RemoteShaclSailValidationException(
+								new StringReader(errInfo.toString()), "", RDFFormat.NQUADS));
+
```

Comment:
I'm trying out throwing a custom exception based based on if the correct header is set to indicate that this is a shacl violation.

Reply:
I would prefer to throw the real ShaclSailValidationException, but doing so requires a dependency on the shacl module which causes a dependency cycle. I need to extract some interfaces instead....which maven model should I use for those interfaces do you think @jeenbroekstra ?

Reply:
If you want a place to have a shared exception class, rdf4j-util might be a good spot. Create a package org.eclipse.rdf4j.common.exception perhaps?